test/acceptance/authz/auto_tenancy_test.go

```diff
@@ -0,0 +1,280 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package authz
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/weaviate/weaviate/client/authz"
+	"github.com/weaviate/weaviate/client/objects"
+	"github.com/weaviate/weaviate/entities/models"
+	"github.com/weaviate/weaviate/grpc/generated/protocol/v1"
+	"github.com/weaviate/weaviate/test/helper"
+	"github.com/weaviate/weaviate/test/helper/sample-schema/articles"
+	"github.com/weaviate/weaviate/usecases/auth/authorization"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+func TestAuthzAutoTenantActivation(t *testing.T) {
+	existingUser := "admin-user"
+	existingKey := "admin-key"
+
+	customUser := "custom-user"
+	customKey := "custom-key"
+
+	testRoleName := "test-role"
+
+	adminAuth := helper.CreateAuth(existingKey)
+
+	_, teardown := composeUp(t, map[string]string{existingUser: existingKey}, map[string]string{customUser: customKey}, nil)
+
+	cls := articles.ParagraphsClass()
+	tenant := "tenant"
+	obj := articles.NewParagraph().WithID("00000000-0000-0000-0000-000000000001").WithTenant(tenant).Object()
+	obj2 := articles.NewParagraph().WithID("00000000-0000-0000-0000-000000000002").WithTenant(tenant).Object()
+
```

Comment:
there are some uuids defined in test/acceptance/authz/helper.go that you can just reuse (UUID1, UUID2,....)